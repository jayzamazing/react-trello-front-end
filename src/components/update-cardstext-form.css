.updateCardsText {
  color: black;
}

.updateCardsText {
  border: unset;
  resize: unset;
  width: 345px;
  height: 75px;
  border-radius: 5px;
}
.updateCardsText:focus, .updateCardsText:active {
  background-color: white;
  border: 1px solid black;
}
