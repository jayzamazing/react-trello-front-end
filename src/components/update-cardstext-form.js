import React from 'react';
import {reduxForm, Field, focus} from 'redux-form';
import Textarea from './textarea';
import {connect} from 'react-redux';
import {updateCards} from '../actions/cards';
import {submitUpdate, blurUpdate} from './utils';
import './update-cardstext-form.css';

export class UpdateCardsTextForm extends React.Component {
  render() {
    return (
      <form className="update-cardstext-area" ref="updatecardstextform"
        onSubmit={
          this.props.handleSubmit(values => {
            return this.props.updateCards(this.props._id,
              values[this.props._id]._id ? values[this.props._id].text : values[this.props._id])
          })}>
          <Field
            format={(data) => {
              return data ? data.text : ''
            }}
            component={Textarea}
            name={this.props._id}
            textareaClass="updateCardsText"
            onKeyPress={e => submitUpdate(e)}
            onBlur={() => blurUpdate(this, 'updatecardstextform')}
          />
      </form>
    );
  }
}
const mapStateToProps = (state, props) => ({
  initialValues: state.cards
});

const mapDispatchToProps = (dispatch) => ({
  //dispatch update to cards name if enter key is pressed
  updateCards: (cardsId, cardsText) => {
    dispatch(updateCards(cardsId, {
      text: cardsText
    }));
  }
});
UpdateCardsTextForm = reduxForm({
    enableReinitialize : true,
    onSubmitFail: (errors, dispatch) =>
        dispatch(focus('updatecards', Object.keys(errors)[0]))
})(UpdateCardsTextForm);
export default connect(mapStateToProps, mapDispatchToProps)(UpdateCardsTextForm);
