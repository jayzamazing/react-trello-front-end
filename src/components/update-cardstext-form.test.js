'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {UpdateCardsTextForm} from './update-cardstext-form';

describe('update-cardstext-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    shallow(<UpdateCardsTextForm handleSubmit={handleSubmit} initialValues={initialValues} _id="35re4qwefgt34werwd"/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const wrapper = shallow(<UpdateCardsTextForm handleSubmit={handleSubmit} initialValues={initialValues} _id="35re4qwefgt34werwd"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on update button and check dispatches
  it('update cardslist click', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const wrapper = shallow(<UpdateCardsTextForm handleSubmit={handleSubmit} initialValues={initialValues} _id="35re4qwefgt34werwd"/>);
    wrapper.find('[name="35re4qwefgt34werwd"]').at(0).simulate('change', { target: { value: 'testing' } });
    wrapper.find('.update-cardstext-area').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
});
