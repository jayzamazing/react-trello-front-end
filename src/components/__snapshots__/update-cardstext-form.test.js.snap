// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update-cardstext-form snapshot 1`] = `
<form
  className="update-cardstext-area"
>
  <Field
    component={[Function]}
    format={[Function]}
    name="35re4qwefgt34werwd"
    onBlur={[Function]}
    onKeyPress={[Function]}
    textareaClass="updateCardsText"
  />
</form>
`;
